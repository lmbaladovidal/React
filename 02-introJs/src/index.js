import heroes from './data/heroes'
console.log(heroes);

const getHeroesById = (id)=>{
    return heroes.find((heroe) => heroe.id === id);
}

const getHeroesByOwner = (owner)=>{
    return heroes.filter((heroe) => heroe.owner === owner);
}

console.log(getHeroesById(2));
console.log(getHeroesByOwner('DC'));
let arr = [0,1,2,3];
arr[10] = 10;
console.log(arr);
console.log(arr[9]);