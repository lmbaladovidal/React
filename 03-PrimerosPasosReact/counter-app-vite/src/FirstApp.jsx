import PropTypes from 'prop-types'
const newMessage = 'Leandro';

function getObj(){
    return {Nombre:"Leandro",
            Apellido:"Balado Vidal"}
}

export const FirstApp = ({title, subtitle,name}) => {
  console.log(title)
  return (
    <>
        <h1 data-testid="test-title">{title}</h1>
        {/* <code>{JSON.stringify(getObj())}</code> */}
        <p>{subtitle}</p>
        <p>{subtitle}</p>
    </>
  )
}

FirstApp.propTypes = {
    title : PropTypes.string.isRequired,
    subtitle : PropTypes.string,
    counter : PropTypes.number
};

FirstApp.defaultProps = {
    name :"Leandro Martin",
    //title : "No Hay titulo",
    subtitle : "No Hay subtitulo"
}
