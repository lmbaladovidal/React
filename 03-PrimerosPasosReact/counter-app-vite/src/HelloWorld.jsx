export const HelloWorld = () => {
  return (
    <h2>HelloWorld</h2>
  )
}
