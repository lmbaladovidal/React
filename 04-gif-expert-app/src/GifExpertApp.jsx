import { useState } from "react";
import { AddCategory } from "./components/AddCategory";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState([ 'Kimetsu no yaiba','My Hero Academy' ]);
  const onAddCategory = (newCategory)=>{
    setCategories([newCategory,...categories]);
  }
  return (
    <>
        {/* titulo */}
        <h1>GifExpertApp</h1>
        {/* input */}
        <AddCategory 
        /* onAddCategory={setCategories} */
        onNewCategory={onAddCategory}
        />
        {/* listado de gif */}
        <ol>          
          {categories.map((category,i)=>{
            return <li key={i}>{category}</li>
          })}
        </ol>
        {/* gif items */}
    </>
  )
}
