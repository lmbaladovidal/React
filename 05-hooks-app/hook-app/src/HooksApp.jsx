import React from 'react'

import { CounterApp } from './01-useState/CounterApp'
import { CounterWithCustomHook } from './01-useState/CounterWithCustomHook'
import { FormWithCustomHook } from './02-useEfect/FormWithCustomHook'
import { SimpleForm } from './02-useEfect/SimpleForm'
import { FocusScreen } from './04-useRef/FocusScreen'

export const HooksApp = () => {
  return (
    <>
        <h1>HookApp</h1>
        <hr/>
{/*         <CounterApp/>
        <CounterWithCustomHook/>
        <SimpleForm/>
        <FormWithCustomHook/> */}
        <FocusScreen/>
    </>
  )
}
