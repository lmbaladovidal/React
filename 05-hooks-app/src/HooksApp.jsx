import React from 'react'

/* import { CounterApp } from './01-useState/CounterApp'
import { CounterWithCustomHook } from './01-useState/CounterWithCustomHook'
import { FormWithCustomHook } from './02-useEfect/FormWithCustomHook'
import { SimpleForm } from './02-useEfect/SimpleForm'
import { MultipleCustomHooks } from './03-Examples/MultipleCustomHooks'
import { FocusScreen } from './04-useRef/FocusScreen'
import { Layout } from './05-useLayoutEffect/LayoutEffect'
import { MemoHook } from './06-Memos/MemoHook'
import { Memorice } from './06-Memos/Memorice' 
import { CallBackHook } from './06-Memos/CallBackHook'
import { Padre } from './07-tarea-memo/Padre'
import './08-useReducer/intro-reducer'*/
import { TodoApp } from './08-useReducer/TodoApp'

export const HooksApp = () => {
  return (
    <>
        <h1>HookApp</h1>
        <hr/>
{/*         <CounterApp/>
        <CounterWithCustomHook/>
        <SimpleForm/>
        <FormWithCustomHook/>
        <MultipleCustomHooks/>
        <FocusScreen/>
        <Layout/>
        <Memorice/>
        <MemoHook/> 
        <CallBackHook/>
        <Padre/>*/}
        <TodoApp/>
    </>
  )
}
