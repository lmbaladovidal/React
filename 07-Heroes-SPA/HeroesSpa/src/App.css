.btn-more {
    text-decoration: none;
    position: relative;
    font-size: 12px;
    text-transform: uppercase;
    text-decoration: none;
    padding: 1em 2.5em;
    display: inline-block;
    border-radius: 6em;
    transition: all .2s;
    border: none;
    font-family: inherit;
    font-weight: 500;
    color: black;
    background-color: white;
   }
   
   .btn-more:hover {
    text-decoration: none;
    transform: translateY(-3px);
    box-shadow: 0 10px 20px rgba(0, 0, 0, 0.2);
   }
   
   .btn-more:active {
    transform: translateY(-1px);
    box-shadow: 0 5px 10px rgba(0, 0, 0, 0.2);
   }
   
   .btn-more::after {
    content: "";
    display: inline-block;
    height: 100%;
    width: 100%;
    border-radius: 100px;
    position: absolute;
    top: 0;
    left: 0;
    z-index: -1;
    transition: all .4s;
   }
   
   .btn-more::after {
    background-color: #fff;
   }
   
   .btn-more:hover::after {
    transform: scaleX(1.4) scaleY(1.6);
    opacity: 0;
   }

   .card{
    margin-top: 10px;
    margin-left: 10px;
    padding-top: 5px;
    transition: 200ms;
    min-width: 300px;
    min-height: 300px;
   }
   .card:hover{
    box-shadow: 0 10px 20px rgba(0, 0, 0, 0.2);
   }