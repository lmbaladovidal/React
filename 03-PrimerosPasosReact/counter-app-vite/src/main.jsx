import React from 'react'
import ReactDOM from 'react-dom/client'
import './styles.css'
import { FirstApp } from './FirstApp';
import {HelloWorld} from './HelloWorld';
import {CounterApp} from './CounterApp';


ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <HelloWorld />
        <CounterApp value={22}/>
        {/* <FirstApp title={"Hola, soy Goku!"} subtitle='Soy un sexy subtitulo' counter={1}/> */}
    </React.StrictMode>
);