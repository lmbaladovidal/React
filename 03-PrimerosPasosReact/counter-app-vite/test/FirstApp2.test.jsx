import {render,screen} from "@testing-library/react"
import { FirstApp } from "../src/FirstApp";
describe('pruebas en <FirstApp/>', () => {
        const title = "Hola, soy Goku";
        const subTitle = "Hola, soy un subtitulo";
        test('debe hacer match con el snapshot', () => { 
                const {container} = render(<FirstApp title={title}/>)
                expect(container).toMatchSnapshot();
        })
        
        test('debe  mostrar el msj "Hola, soy Goku"', () => { 
                render(<FirstApp title={title}/>)
                expect(screen.getByText).toBeTruthy();
                //screen.debug();
         })
         test('debe mostrar el titulo en un h1', () => { 
                render(<FirstApp title={title}/>)
                expect(screen.getByRole('heading', {level:1}).innerHTML).toContain(title);
          });
          test('debe mostar el subtitulo enviado por props', () => { 
                render(<FirstApp title={title} subtitle={subTitle}/>);
                expect(screen.getAllByText(subTitle).length).toBe(2);

           })

    
});