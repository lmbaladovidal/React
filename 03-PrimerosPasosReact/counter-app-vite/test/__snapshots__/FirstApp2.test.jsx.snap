// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <FirstApp/> debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola, soy Goku
     
  </h1>
  <p>
    No Hay subtitulo
  </p>
  <p>
    No Hay subtitulo
  </p>
</div>
`;
